inst/include/dust/dust.hpp

```diff
@@ -282,15 +283,45 @@ class Dust {
   // contents of the particle state (uses the set_state() and swap()
   // methods on particles).
   void reorder(const std::vector<size_t>& index) {
+#ifdef _OPENMP
+    #pragma omp parallel for schedule(static) num_threads(_n_threads)
+#endif
     for (size_t i = 0; i < _particles.size(); ++i) {
       size_t j = index[i];
       _particles[i].set_state(_particles[j]);
     }
-    for (auto& p : _particles) {
-      p.swap();
+#ifdef _OPENMP
+    #pragma omp parallel for schedule(static) num_threads(_n_threads)
+#endif
+    for (size_t i = 0; i < _particles.size(); ++i) {
+      _particles[i].swap();
     }
   }
 
+  std::vector<size_t> resample(const std::vector<real_t>& weights) {
```

Comment:
Can we add a boolean `log` argument here so that this can work with log weights, where we add the scale_log_weights() function which deals with the log-sum-exp issue? I'm sure this function was going to be added when the full particle filter implementation appears soon, but seems like it might be handy to have this as an option here too

Reply:
I'm voting to hold off on this for now - we can add it if we need it